pyproject.toml

```diff
@@ -34,3 +34,55 @@ exclude = '''
      )/
 )
 '''
+
+# these four lines are needed by poetry to work
+[tool.poetry]
+name = "xonsh"
+version = "0.0.0" # check the setup.py file
+description = ""
+authors = []
+
+[tool.poetry.dependencies]
+python = ">=3.6.1,<4.0"
+
+[tool.poetry.dev-dependencies]
+# tests deps
+pytest = "^6.1.1"
```

Comment:
Do most of these things need pinnings? It seems to me like they don't

Reply:
it is the convention to have these pinned. everytime `poetry update` it updates the lockfile to be exactly the same. You could check the projects using poetry. Most of the time it is important to have a correct set of working versions.

Reply:
Sure it just seems to be a little odd to have them pinned so tightly without knowing that the patch numbers are actually required.  Also, I'd like to avoid caret and tilde pins since conda does not understand them. (Also I personally find them harder to read.)  For example, this seems like it should be `>=6` or `>=6,<7`

Reply:
I guess maybe I don't fully understand what poetry --export does, either

Reply:
It would be better if you could give it a try. Just going through its quickstart. The best part about using it having the same environment in all the systems.

Reply:
> I guess maybe I don't fully understand what poetry --export does, either

it returns the dependencies in the lock file in requirements.txt format. I have used it in the rever.xsh 
to retain the original functionality.

Reply:
Ok yeah revert was the big concern